modules/text/src/text_detector_swt.cpp

```diff
@@ -0,0 +1,854 @@
+    }
+    attributes.mean = sum / ((float)component.size());
+    for (size_t i = 0; i < component.size(); i++) {
+        attributes.variance += (temp[i] - attributes.mean) * (temp[i] - attributes.mean);
+    }
+
+    attributes.variance = attributes.variance / ((float)component.size());
+    std::sort(temp.begin(),temp.end());
+    attributes.median = temp[temp.size()/2];
+
+    attributes.length = (float) (attributes.xmax - attributes.xmin + 1);
+    attributes.width = (float) (attributes.ymax - attributes.ymin + 1);
+    return attributes;
+}
+
+void renderComponents (const Mat& SWTImage, const std::vector<Component>& components, Mat& output)
+{
+    output.setTo(0);
+
+    for (size_t i = 0; i < components.size(); i++) {
+        const Component& component = components[i];
+        for (size_t j = 0; j < component.points.size(); j++)
+        {
+            const SWTPoint& pt = component.points[j];
+            output.at<float>(pt.y, pt.x) = SWTImage.at<float>(pt.y, pt.x);
+        }
+    }
+    for(int row = 0; row < output.rows; row++ ){
+        float* ptr = output.ptr<float>(row);
+        for ( int col = 0; col < output.cols; col++ ){
+            if (*ptr == 0) {
+                *ptr = -1;
+            }
+            ptr++;
+        }
+    }
+    float maxVal = 0;
+    float minVal = (float) FLT_MAX;
+    for(int row = 0; row < output.rows; row++ ){
+        const float* ptr = output.ptr<float>(row);
+        for ( int col = 0; col < output.cols; col++ )
+        {
+            float v = ptr[col];
+            if (v != 0)
+            {
+                maxVal = std::max(*ptr, maxVal);
+                minVal = std::min(*ptr, minVal);
+            }
+        }
+    }
+    float difference = maxVal - minVal;
+    for(int row = 0; row < output.rows; row++ ) {
+        float* ptr = output.ptr<float>(row);
+        for (int col = 0; col < output.cols; col++)
+        {
+            float& v = ptr[col];
+            if (v < 1) {
+                v = 1;
+            } else {
+                v = (v - minVal)/difference;
+            }
+        }
+    }
+
+}
+
+std::vector<Component> filterComponents(const Mat& SWTImage, const std::vector<std::vector<SWTPoint>>& components, bool skipChecks)
+{
+    const int NUM_THETA = 36;  // in 180 (CV_PI)
+
+    std::vector<Component> filteredComponents;
+    filteredComponents.reserve(components.size());
+    for (size_t i = 0; i < components.size(); i++)
+    {
+        const vector<SWTPoint>& component = components[i];
+        ComponentAttr attributes = getAttributes(component, SWTImage);
+        if (!skipChecks && attributes.variance > 0.5 * attributes.mean) continue;
+        if (!skipChecks && attributes.width > 300) continue;
+
+
+        float area = attributes.length * attributes.width;
+
+        // compute the rotated bounding box
+        for (int theta_i = 1; theta_i < (NUM_THETA / 2); theta_i++)
+        {
+            float theta = (float)i * (CV_PI / NUM_THETA);
```

Comment:
Replaced for loop with floating-point iterator to integer one (floating-point is not accuracy between platforms and should be avoided).
Please check loop boundary (`<` pr `<=`) and fix if necessary.

Old code:
```
float increment = ((float) 1. )/ (float) 36.;
for (float theta = increment * (float) CV_PI; theta<CV_PI/2.0; theta += increment * (float) CV_PI) {
```

Reply:
The `<` looks good. But in line 446, it should `(float) theta_i * (CV_PI / NUM_THETA)` and not  `(float)i * (CV_PI / NUM_THETA)`. I will correct that and go over the rest of the code.